serdect: safer bytes handling (#1112)
RustCrypto/formats: serdect/README.md

other kinds of data-dependent branching on the contents of the serialized data,
using a constant-time hex serialization with human-readable formats should
help reduce the overall timing variability.

> The table below lists the formats `serdect` is tested against, and
> whether the serialization via `array` or `slice` is constant-time for them.

Reviewer: I think we should be careful when using the word "constant-time", as pointed out above, this is best effort, we can't actually guarantee it.
```suggestion
The table below lists the formats `serdect` is tested against, and
whether the serialization via `array` or `slice` is data-independent.
```

Author: That whole paragraph is largely a placeholder, if someone can suggest a better wording for it, I will happily accept.

Author: Rephrased.

> | Format                          | `array`  | `slice`  |
> |---------------------------------|----------|----------|
> | Bincode (`bincode=^1`)          | &#x2705; | &#x2705; |
> | CBOR (`ciborium=^0.2`)          | &#x274c; | &#x2705; |
> | MessagePack (`rmp-serde`)       | &#x274c; | &#x2705; |
> | JSON (`serde-json-core=^0.5`)   | &#x2705; | &#x2705; |
> | JSON (`serde-json=^1`)          | &#x2705; | &#x2705; |
> | TOML (`toml=^0.7`)              | &#x2705; | &#x2705; |

Reviewer: Why is `rmp-serde` missing a verrsion? Also I don't think it makes sense to talk about the individual formats, we should just say which crate, because we don't actually test the formats, but only their implementations.

I also added links to each crate and centered the other columns.
```suggestion
| Format                                                             | `array`  | `slice`  |
|--------------------------------------------------------------------|:--------:|:--------:|
| [`bincode`](https://crates.io/crates/bincode) v1                   | &#x2705; | &#x2705; |
| [`ciborium`](https://crates.io/crates/ciborium) v0.2               | &#x274c; | &#x2705; |
| [`rmp-serde`](https://crates.io/crates/rmp-serde) v0.2             | &#x274c; | &#x2705; |
| [`serde-json-core`](https://crates.io/crates/serde-json-core) v0.5 | &#x2705; | &#x2705; |
| [`serde-json`](https://crates.io/crates/serde-json) v1             | &#x2705; | &#x2705; |
| [`toml`](https://crates.io/crates/toml) v0.7                       | &#x2705; | &#x2705; |
```

Author: > Why is rmp-serde missing a verrsion? 

Forgot to add. Fixed now, and added the links as well.

> 

Reviewer: I think we are good to go to fix arrays in this PR as well, please correct me if I'm wrong @tarcieri, so this table could probably just list crates that we test against.
